components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,320 @@
+			for _, rewardValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				rewards[prizeEnum .. index] = prizeDatum.rowParse(self, rewardValue, args)
+			end
+		end
+	end
+
+	return rewards
+end
+
+function PrizePool:_readSpecialData(args)
+	local data = {}
+
+	for enum in pairs(specialDataTypes) do
+		local type = specialData[specialDataTypes[enum]]
+		local fieldName = type.field
+		if args[fieldName] then
+			data[enum] = type.parse(self, args[fieldName], args)
+		end
+	end
+
+	return data
+end
+
+function PrizePool:_readPlacements(args)
+	local placements = {}
+
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		local placementInput = Json.parseIfString(args[placementIndex])
+		if not placementInput then
+			break
+		end
+
+		local placement = {}
+		-- Parse place, explicit
+		if placementInput.place then
+			local places = Table.mapValues(mw.text.split(placementInput.place, '-'), mw.text.trim)
+			places = Table.mapValues(places, tonumber)
+			placement.placeStart = places[1]
+			placement.placeEnd = places[2] or places[1]
+		end
+
+		-- Parse prizes
+		placement.prizes = self:_readPrizeRewards(placementInput)
+
+		-- Parse opponents in the placement
+		placement.opponents = {}
+		for opponentIndex = 1, math.huge do
+			local opponentInput = Json.parseIfString(placementInput[opponentIndex])
+			if not opponentInput then -- TODO: always iterate until the end placeEnd, if given
+				break
+			end
+
+			local opponent = {opponentData = {}, prizes = {}, data = {}}
+
+			-- Parse Opponent Data
+			opponentInput.type = opponentInput.type or self.opponentType
+			opponent.opponentData = self:_parseOpponentArgs(opponentInput, opponentInput.date)
+
+			-- Parse special prizes for this opponent
+			opponent.prizes = self:_readPrizeRewards(opponentInput)
+
+			-- Parse additional data (groupscore, last opponent etc)
+			opponent.data = self:_readSpecialData(opponentInput)
+
+			-- Set the date
+			opponent.date = opponentInput.date
+
+			table.insert(placement.opponents, opponent)
+		end
+
+		-- Parse place, implicit
+		if not placementInput.place then
+			placement.placeStart = currentPlace + 1
+			placement.placeEnd = currentPlace + #placement.opponents
+		end
+
+		assert(placement.placeStart and placement.placeEnd, 'readPlacements: Invalid |place= provided.')
+		assert(#placement.opponents > placement.placeEnd - placement.placeStart, 'readPlacements: Too many opponents')
+		currentPlace = placement.placeEnd
+		table.insert(placements, placement)
+	end
+
+	return placements
+end
+
+function PrizePool:_parseOpponentArgs(input, date)
+	local opponentArgs = Json.parseIfTable(input) or (type(input) == 'table' and input or {input})
+	opponentArgs.type = opponentArgs.type or self.opponentType
+	opponentArgs.date = date or self.date
+	assert(Opponent.isType(opponentArgs.type), 'Invalid type')
+	local opponentData = Opponent.readOpponentArgs(opponentArgs) or Opponent.tbd()
```

Comment:
```suggestion
	local opponentData = Opponent.readOpponentArgs(opponentArgs, date) or Opponent.tbd()
```
needed for non team opponents to be able to resolve the player teams (see #1390)